### PR TITLE
Mark as spam

### PR DESCRIPTION
**Pull request by sawirricardo**
Referencing the issue #507 
- User can mark a thread as spam
- Notify moderators after 3 or more user report it as spam

### CODE REVIEW

File: app/Notifications/ThreadMarkedAsSpamNotification.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use Illuminate\Notifications\Notification;
+
+class ThreadMarkedAsSpamNotification extends Notification
+{
+    use Queueable;
+
+    public function __construct(
+        private Thread $thread
+    ) {
+    }
```
Comment (driesvints) [inline app/Notifications/ThreadMarkedAsSpamNotification.php:17]:
```suggestion
    public function __construct(private Thread $thread)
    {
    }
```

**Review by driesvints — CHANGES_REQUESTED** [review]
Heya, thanks for your PR! Can you do these changes and rebase with `main` branch?

File: app/Jobs/MarkThreadAsSpam.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+use App\Models\User;
+use App\Notifications\ThreadMarkedAsSpamNotification;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Notification;
+
+class MarkThreadAsSpam implements ShouldQueue
```
[the remark this answers is not in the document]
Comment (sawirricardo) [reply app/Jobs/MarkThreadAsSpam.php]: Resolved with commit https://github.com/laravelio/laravel.io/pull/878/commits/626a53bce3b343b489158a3f346a1194d186f3de

**Comment by driesvints** [comment]
Seems like rebasing went wrong here. Can you resend this in?

**Pull request closed without merging**